### PR TITLE
Clean up some of the linting warnings and errors

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

Clean up linting warnings and errors in the Alloy docs. Some issues (especially passive voice) remain in the various topics. It will take multiple passes to get the warnings/errors down to a manageable level.  

Changes here include:

* Fix some spelling
* Add contractions
* Delete duplicate links
* Delete unused links
* Fix broken or incorrect links
* Fix up some of the passive voice and replace with more active statements
* Tidy up some of the markdown style
* Clean up some word use
* Replace smart quotes
* Remove trailing spaces
* Remove extra blank lines between elements
* Add missing trailing line at the end of the file
* Add missing language to codefences
* Add/remove spaces before/after codefences
* Add/remove spaces before/after lists
* Fix spacing/indentation
* Make bullets `*` for consistency
* Remove gerunds from headings
* Clean up use of `let's` and `we`
* Fix a spelling mistake in a file name and add an alias

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Comment by clayton-cornell** [comment]
The linting cleanup isn't complete, but this PR is big enough as it is. Time for a review/merge.

File: docs/sources/collect/choose-component.md [hunk 1]
```diff
@@ -17,10 +17,9 @@ The components you select and configure depend on the telemetry signals you want
 ## Metrics for infrastructure
 
 Use `prometheus.*` components to collect infrastructure metrics.
-This will give you the best experience with [Grafana Infrastructure Observability][].
+This gives you the best experience with [Grafana Infrastructure Observability][].
```
Comment (BeverlyJaneJ) [inline docs/sources/collect/choose-component.md:20]: empty brackets are okay here?

File: docs/sources/collect/datadog-traces-metrics.md [hunk 2]
```diff
@@ -139,19 +139,19 @@ We recommend this approach for current Datadog users who want to try using {{< p
...
 You can do this by setting up your Datadog Agent in the following way:
 
 1. Replace the DD_URL in the configuration YAML:
 
    ```yaml
     dd_url: http://<DATADOG_RECEIVER_HOST>:<DATADOG_RECEIVER_PORT>
    ```
-Or by setting an environment variable:
 
+Or by setting an environment variable:
```
Comment (BeverlyJaneJ) [inline docs/sources/collect/datadog-traces-metrics.md:154]: should this section be indented under the step number?

File: docs/sources/collect/metamonitoring.md [hunk 3]
```diff
@@ -119,7 +120,7 @@ loki.write "default" {
 
 The [tracing][] block defines the tracing behavior of {{< param "PRODUCT_NAME" >}}.
 
-In this task you will use the [tracing][] block to forward {{< param "PRODUCT_NAME" >}} internal traces to a compatible component. The block is specified without a label and can only be provided once per configuration file.
+In this task you use the [tracing][] block to forward {{< param "PRODUCT_NAME" >}} internal traces to a compatible component. The block is specified without a label and can only be provided once per configuration file.
```
Comment (BeverlyJaneJ) [inline docs/sources/collect/metamonitoring.md:123]:
```suggestion
In this task, you use the [tracing][] block to forward {{< param "PRODUCT_NAME" >}} internal traces to a compatible component. The block is specified without a label and can only be provided once per configuration file.
```

**Review by BeverlyJaneJ — APPROVED** [review]

File: docs/sources/collect/choose-component.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/collect/choose-component.md:20]: The link is defined on line 27, so it should be OK

File: docs/sources/collect/datadog-traces-metrics.md (hunk 2 above)
Comment (clayton-cornell) [inline docs/sources/collect/datadog-traces-metrics.md:154]:
```suggestion
   Or by setting an environment variable:
```

Comment (clayton-cornell) [reply docs/sources/collect/datadog-traces-metrics.md:154]: I think so yes.

Comment (clayton-cornell) [reply docs/sources/collect/datadog-traces-metrics.md]: Fixed in the last commit

**Pull request merged**